contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -4453,3 +4464,58 @@ transform_pivot_clause(ParseState *pstate, SelectStmt *stmt)
 	tsql_outmost_estat->pivot_number++;	
 	MemoryContextSwitchTo(oldContext);
 }
+
+static Node* optimize_explicit_cast(ParseState *pstate, Node *node)
+{
+	if (sql_dialect != SQL_DIALECT_TSQL)
+		return node;
+	if (node == NULL)
+		return NULL;
+	if (IsA(node, FuncExpr))
+	{
+		FuncExpr   *f = (FuncExpr *) node;
+		if (f->funcformat == COERCE_EXPLICIT_CAST){
```

Comment:
Please double check this `funcformat` is only for `CAST` function and no other functions, e.g., UDF, can have this.